php-frontend/src/main/java/org/sonar/php/metrics/CognitiveComplexityVisitor.java

```diff
@@ -0,0 +1,318 @@
+    visitWithNesting(()-> super.visitFunctionDeclaration(tree));
+  }
+
+  @Override
+  public void visitFunctionExpression(FunctionExpressionTree tree) {
+    visitWithNesting(()-> super.visitFunctionExpression(tree));
+  }
+
+  @Override
+  public void visitMethodDeclaration(MethodDeclarationTree tree) {
+    visitWithNesting(()-> super.visitMethodDeclaration(tree));
+  }
+
+  @Override
+  public void visitConditionalExpression(ConditionalExpressionTree tree) {
+    complexity.addComplexityWithNesting(tree.queryToken());
+    visit(tree.condition());
+    visitWithNesting(tree.trueExpression());
+    visitWithNesting(tree.falseExpression());
+  }
+
+  @Override
+  public void visitBreakStatement(BreakStatementTree tree) {
+    if (tree.argument() != null) {
+      complexity.addComplexityWithoutNesting(tree.breakToken());
+    }
+    super.visitBreakStatement(tree);
+  }
+
+  @Override
+  public void visitContinueStatement(ContinueStatementTree tree) {
+    if (tree.argument() != null) {
+      complexity.addComplexityWithoutNesting(tree.continueToken());
+    }
+    super.visitContinueStatement(tree);
+  }
+
+  @Override
+  public void visitGotoStatement(GotoStatementTree tree) {
+    complexity.addComplexityWithoutNesting(tree.gotoToken());
+    super.visitGotoStatement(tree);
+  }
+
+  @Override
+  public void visitBinaryExpression(BinaryExpressionTree tree) {
+    if (tree.is(CONDITIONAL_AND, CONDITIONAL_OR) && !nestedLogicalExpressions.contains(tree)) {
+      List<SyntaxToken> flatOperators = new ArrayList<>();
+      flattenLogicalExpression(0, flatOperators, tree);
+
+      complexity.addComplexityWithoutNesting(flatOperators.get(0));
+
+      for (int i = 1; i < flatOperators.size(); i++) {
+        if (!flatOperators.get(i).text().equals(flatOperators.get(i - 1).text())) {
+          complexity.addComplexityWithoutNesting(flatOperators.get(i));
+        }
+      }
+    }
+
+    super.visitBinaryExpression(tree);
+  }
+
+  private void flattenLogicalExpression(int i, List<SyntaxToken> operators, ExpressionTree expression) {
+    if (expression.is(CONDITIONAL_AND, CONDITIONAL_OR)) {
+      nestedLogicalExpressions.add(expression);
+
+      BinaryExpressionTree binaryExpression = (BinaryExpressionTree) expression;
+      operators.add(i, binaryExpression.operator());
+
+      ExpressionTree leftChild = removeParenthesis(binaryExpression.leftOperand());
+      ExpressionTree rightChild = removeParenthesis(binaryExpression.rightOperand());
+
+      flattenLogicalExpression(i + 1, operators, rightChild);
+      flattenLogicalExpression(i, operators, leftChild);
+    }
+  }
+
+  private void visitWithNesting(@Nullable Tree tree) {
+    if (tree != null) {
+      visitWithNesting(() -> tree.accept(this));
+    }
+  }
+
+  private void visitWithNesting(List<StatementTree> statements) {
+    visitWithNesting(() ->
+      statements.forEach(statementTree -> statementTree.accept(this)));
+  }
+
+  private void visitWithNesting(Runnable action) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused private "visitWithNesting" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUnusedPrivateMethod)